ag/validation.go

```diff
@@ -49,6 +52,23 @@ func Interceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
 		}
 		ctx, cancel := context.WithTimeout(ctx, MaxWait)
 		defer cancel()
+
+		meta, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			return nil, errors.New("Could not grab metadata from context")
+		}
+
+		if len(meta.Get("cookie")) > 0 {
```

Comment:
Do this only once:
```golang
token := meta.Get(Cookie)
```
Define `Cookie` as a const. 

Then you can do `if len(token) > 0` and save a lookup in meta.